Doc/whatsnew/3.14.rst

```diff
@@ -794,6 +795,18 @@ For further information on how to build Python, see
 (Contributed by Ken Jin in :gh:`128563`, with ideas on how to implement this
 in CPython by Mark Shannon, Garrett Gu, Haoran Xu, and Josh Haberman.)
 
+.. _whatsnew314-free-threaded-cpython:
+
+Free-threaded mode
+---------------------
+
+Free-threaded mode (:pep:`703`), initially added in 3.13, has been significantly improved. The implementation described in PEP 703 was finished, including C API changes, and temporary workarounds in the interpreter were replaced with more permanent solutions. The specializing adaptive interpreter (:pep:`659`) is now enabled in free-threaded mode, which along with many other optimizations greatly improves its performance. The performance penalty on single-threaded code in free-threaded mode is now roughly 5-10%, depending on platform and C compiler used.
+
+This work was done by many contributors: Sam Gross, Matt Page, Neil Schemenauer, Thomas Wouters, Donghee Na, Kirill Podoprigora, Ken Jin, Itamar Oren, Brett Simmers, Dino Viehland, Nathan Goldbaum, Ralf Gommers,
+Lysandros Nikolaou, and many others.
```

Comment:
I think we should include Kumar Aditya in the list of contributors as well.

```suggestion
This work was done by many contributors: Sam Gross, Matt Page, Neil Schemenauer, Thomas Wouters, Donghee Na, Kirill Podoprigora, Ken Jin, Itamar Oren, Brett Simmers, Dino Viehland, Nathan Goldbaum, Ralf Gommers,
Lysandros Nikolaou, Kumar Aditya, and many others.
```

Reply:
We're forgetting Edgar Margffoy as well.